server/Arcadia.Assistant.CSP/CspCalendarEventsApprovalsChecker.cs

```diff
@@ -40,7 +76,7 @@ protected override async Task<string> GetNextApprover(string employeeId, IEnumer
 
                 finalApprover = parentDepartment?.ChiefId;
             }
-            else if (employeeDepartment.IsHeadDepartment && employeeDepartment.ChiefId == employeeId)
+            else if (employeeDepartment.IsHeadDepartment && isEmployeeChief)
```

Comment:
do you also need to return null in here?

Reply:
No, it is handled below.